django/db/models/fields/utils.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+
+
+class FieldCacheMixin(object):
+    """
+    Provides a common API for working with the fields
+    value cache on an instance.
+    """
+
+    def get_cache_name(self):
```

Comment:
So I've gone deeper into it and basically, the reason why this never raised an error is because `self.name` on `ForeignObjectRel` calls `related_query_name` which uses related name or the model name and `get_accessor_name` does the same except that when it's called from a many to one or many to many it adds the `_set` to the default accessor if none is specified using `related_name`.

The cache on the reverse side is only called if the object is a one to one relationship and so in this case `self.name` and `self.get_accessor_name()` will always be the same value.

If it's ok with you I'd like to use `get_accessor_name()` as was the default before, the current tests cover this case already so there's no need to add any as `get_cache_name()` is only called in certain circumstances.